app/src/main/java/com/github/sdp_begreen/begreen/fragments/SendPostFragment.kt

```diff
@@ -189,11 +199,69 @@ class SendPostFragment : Fragment() {
+                val connectedRef = Firebase.database.getReference(".info/connected")
+
+                // Attach a listener for a single event to check if client is currently connected to Firebase.
+                connectedRef.addListenerForSingleValueEvent(object : ValueEventListener {
+                    override fun onDataChange(snapshot: DataSnapshot) {
+                        // Retrieve connectivity state from snapshot. If data is null, assume not connected.
+                        val connected = snapshot.getValue(Boolean::class.java) ?: false
+
+                        // If connected, proceed with user update and photo sharing operations.
+                        if (connected) {
+                            Log.d("TAG", "connected")
+                            // Launch coroutine to perform potentially long-running operations off the main UI thread.
+                            lifecycleScope.launch {
+                                // Update the user with the new photo and return to the camera.
+                                updateUser(metadata, user)
+                                returnToCamera()
+                            }
+                        // If not connected, store metadata and user information for later use when connectivity is re-established.
+                        } else {
+                            Log.d("TAG", "not connected")
+
+                            // Use TinyDB for local data storage.
+                            val tinyDB = TinyDB(this@SendPostFragment.context)
+
+                            // Retrieve existing metadata and add current one.
+                            val metas = tinyDB.getListObject("metas", TrashPhotoMetadata::class.java)
+                            metas.add(metadata!!)
+                            tinyDB.putListObject("metas", metas)
+
+                            // Retrieve existing user data and add current user.
+                            val users = tinyDB.getListObject("users", User::class.java)
+                            users.add(user)
+                            tinyDB.putListObject("users", users)
+
+                            view?.findViewById<ImageView>(R.id.preview)?.drawable?.toBitmap()?.also { bitmap ->
+
+                                // Get the stored metadata
+                                val baos = ByteArrayOutputStream()
+                                bitmap.compress(Bitmap.CompressFormat.PNG,100,baos)
+
+                                // Convert the bitmap to a byte array and then to a base64 string for easier storage.
+                                val b: ByteArray = baos.toByteArray()
+                                val encoded: String = Base64.encodeToString(b, Base64.DEFAULT)
+                                val bitmaps = tinyDB.getListString("bitmaps")
+                                bitmaps.add(encoded)
+                                tinyDB.putListString("bitmaps", bitmaps)
```

Comment:
Here, instead of encoding the whole image to String, to retrieve it later, I would suggest to have a closer look at how Xavier passed the image from the `CameraWithUIFragment` to the `SendPostFragment`. He is already storing the taken picture on the phone, and he is using its `uri` to get it. He is doing that in the method `InitView`.

And then I would suggest you to only store the URI, and then you will be able to retrieve the picture later on when the connection is back.